### ISSUE TITLE
Libraries do not get uninstalled

### ISSUE THREAD

**Issue by jgalan** [rest-for-physics/framework#445]
I have disabled a library from the compilation, 

E.g. `cmake -DRESTLIB_AXION=OFF ..`

However, when I execute `restRoot` in the new installed version of REST the library is still loading.

I believe the uninstall system is not working with `.so` installed libraries?

**Comment by juanangp** [comment]
I cannot reproduce the issue, while doing `make uninstall` all the libraries are properly uninstalled.

Perhaps this is a corner case when you do `cmake -DRESTLIB_AXION=OFF ...` in the same build? In this case would be better to call `make uninstall` before doing  `cmake`.

**Comment by jgalan** [comment]
Yes, may be, it couldn't be that the `cmake` detects that I used `-DRESTLIB_AXION=OFF` and calls `make uninstall`? Or that whenever we invoke `cmake` the `make uninstall` is invoked?

**Comment by juanangp** [comment]
> Yes, may be, it couldn't be that the `cmake` detects that I used `-DRESTLIB_AXION=OFF` and calls `make uninstall`? Or that whenever we invoke `cmake` the `make uninstall` is invoked?

We invoke `make uninstall` whenever `make install` is invoked, we cannot do at `cmake` because the `install_manifest` might not be generated at this stage.

In fact, I think the problem is in the `cmake` itself since I think that axionlib still gets installed after doing  `cmake -DRESTLIB_AXION=ON ...` and then `cmake -DRESTLIB_AXION=OFF ...`
